src/module-elasticsuite-tracker/Observer/CleanCookiesOnLogout.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * DISCLAIMER
+ * Do not edit or add to this file if you wish to upgrade Smile Elastic Suite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteTracker
+ * @author    Romain Ruaud <romain.ruaud@smile.fr>
+ * @copyright 2017 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+namespace Smile\ElasticsuiteTracker\Observer;
+
+/**
+ * Customer Logout observer : used to clean tracking cookies on customer log out.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteTracker
+ * @author   Romain Ruaud <romain.ruaud@smile.fr>
+ */
+class CleanCookiesOnLogout implements \Magento\Framework\Event\ObserverInterface
+{
+    /**
+     * @var \Smile\ElasticsuiteTracker\Helper\Data
+     */
+    private $helper;
+
+    /**
+     * @var \Magento\Framework\Stdlib\CookieManagerInterface
+     */
+    private $cookieManager;
+
+    /**
+     * @var \Magento\Framework\Stdlib\Cookie\CookieMetadataFactory
+     */
+    private $cookieMetadataFactory;
+
+    /**
+     * CleanCookiesOnLogout constructor.
+     *
+     * @param \Smile\ElasticsuiteTracker\Helper\Data                 $helper                Elasticsuite tracking helper.
+     * @param \Magento\Framework\Stdlib\CookieManagerInterface       $cookieManager         Cookie Manager.
+     * @param \Magento\Framework\Stdlib\Cookie\CookieMetadataFactory $cookieMetadataFactory Cookie Metadata Factory.
+     */
+    public function __construct(
+        \Smile\ElasticsuiteTracker\Helper\Data $helper,
+        \Magento\Framework\Stdlib\CookieManagerInterface $cookieManager,
+        \Magento\Framework\Stdlib\Cookie\CookieMetadataFactory $cookieMetadataFactory
+    ) {
+        $this->helper                = $helper;
+        $this->cookieManager         = $cookieManager;
+        $this->cookieMetadataFactory = $cookieMetadataFactory;
+    }
+
+    /**
+     * Clean customer tracking cookies when logging out.
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     *
+     * @event customer_logout
+     *
+     * @param \Magento\Framework\Event\Observer $observer The observer
+     *
+     * @return void
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $cookieNames = $this->getCookies();
+
+        $metadata = $this->cookieMetadataFactory->createPublicCookieMetadata();
+        $metadata->setPath('/');
```

Comment:
What about site hosted under a different path ? Bug ?

Reply:
Indeed, it may be a bug, but probably in the tracker himself because it is actually enforcing the cookie path on "/" : https://github.com/Smile-SA/elasticsuite/blob/master/src/module-elasticsuite-tracker/view/frontend/web/js/tracking.js#L35

Reply:
Should be fixed for both.

Reply:
ok for me, I was not sure if the arbitrary "/" on tracker was by design or a bug. I will use default magento cookie path for the tracker so this should work server/client side in the same way.